netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -595,17 +607,21 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
       NettyClientHandler.this.onRstStreamRead(streamId, errorCode);
     }
 
-    @Override public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data)
-        throws Http2Exception {
+    @Override
+    public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2Exception {
       Http2Ping p = ping;
-      if (p != null) {
+      if (data.getLong(data.readerIndex()) == flowControlPing.payload()) {
```

Comment:
This comment was collapsed, but still applies:
> Since flowControlPing is hard-coded, it can collide with the randomly-generated ping id in sendPingFrame. I think we can just hard-code it instead of choosing a random value each time (since it already guarantees only one is in progress at a time).